improver_tests/utilities/test_cube_checker.py

```diff
@@ -318,5 +322,99 @@ def test_unmatching_y(self):
+            datetime(2017, 11, 10, 3, 0),
+            datetime(2017, 11, 10, 4, 0),
+        )
+    cube = set_up_variable_cube(data, **kwargs)
+    for descriptor in (
+        {"name": "air_temperature", "units": "K"},
+        {"name": "air_pressure", "units": "Pa"},
+        {"name": "relative_humidity", "units": "kg kg-1"},
+    ):
+        cube = cube.copy()
+        cube.rename(descriptor["name"])
+        cube.units = descriptor["units"]
+        cubes.append(cube)
+    return cubes
+
+
+@pytest.mark.parametrize("time_bounds", (True, False))
+@pytest.mark.parametrize("input_count", (2, 3))
+def test_time_coords_valid(cubes, input_count, time_bounds: bool):
+    """Test that no exceptions are raised when the required conditions are met
+    for either 2 or 3 cubes, with or without time bounds"""
+    assert_time_coords_valid(cubes[:input_count], time_bounds=time_bounds)
+
+
+def inconsistent_time_bounds(cubes: List[Cube]):
+    """Adds time bounds only to the first cube"""
+    time_point = cubes[0].coord("time").points[0]
+    cubes[0].coord("time").bounds = (time_point - 3600, time_point)
+
+
+def inconsistent_time_point(cubes: List[Cube]):
+    """Moves time point of first cube back by one hour"""
+    cubes[0].coord("time").points = cubes[0].coord("time").points - 3600
+
+
+def inconsistent_frt(cubes: List[Cube]):
+    """Moves forecast_reference_time point of first cube back by one hour"""
+    cubes[0].coord("forecast_reference_time").points = (
+        cubes[0].coord("forecast_reference_time").points - 3600
+    )
+
+
+def remove_one_time_bounds(cubes: List[Cube]):
+    """Removes time bounds from first cube"""
+    cubes[0].coord("time").bounds = None
+
+
+def remove_two_time_bounds(cubes: List[Cube]):
+    """Removes time bounds from first two cubes"""
+    cubes[0].coord("time").bounds = None
+    cubes[1].coord("time").bounds = None
+
+
+def only_one_cube(cubes: List[Cube]):
+    """Removes second and third cubes"""
+    cubes.pop(2)
+    cubes.pop(1)
+
+
+@pytest.mark.parametrize(
+    "modifier, time_bounds, error_match",
+    (
+        (inconsistent_time_bounds, False, "air_temperature must not have time bounds"),
+        (inconsistent_time_point, False, "time coordinates do not match."),
+        (inconsistent_frt, False, "forecast_reference_time coordinates do not match."),
+        (remove_one_time_bounds, True, "air_temperature must have time bounds"),
+        (
+            remove_two_time_bounds,
+            True,
+            "air_temperature and air_pressure must have time bounds",
+        ),
+        (only_one_cube, False, "Need at least 2 cubes to check. Found 1",),
+    ),
+)
+def test_time_coord_exceptions(
+    cubes, modifier: callable, time_bounds: bool, error_match: str
+):
+    """Checks that assert_time_coords_valid raises useful errors
+    when the required conditions are not met."""
+    for cube in cubes:
+        cube.attributes["mosg__model_configuration"] = "gl_ens"
```

Comment:
Is there a reason to set this up like this, or could you a) not have this attribute or b) pass it into `set_up_variable_cube`?

Reply:
I think I copied these from another test. This attribute is entirely unnecessary for these tests, so I've deleted these two lines.